src/App.tsx

```diff
@@ -1,39 +1,69 @@
-import React from 'react';
-import './App.scss';
+import { Component } from 'react';
+import { Clock } from './Clock';
+
+interface State {
+  clockName: string,
+  hasClock: boolean,
+}
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+export class App extends Component<{}, State> {
+  state = {
+    hasClock: true,
+    clockName: 'Clock-0',
+  };
+
+  timerId = Number();
+
+  componentDidMount() {
+    document.addEventListener('contextmenu', this.hideClock);
+    document.addEventListener('click', this.showClock);
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+    this.timerId = window.setInterval(() => {
+      this.setState({
+        clockName: getRandomName(),
+      });
+    }, 3300);
+  }
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+  componentDidUpdate(_: never, prevState: State) {
+    const { clockName, hasClock } = this.state;
+    const { clockName: prevClockName } = prevState;
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+    if (clockName !== prevClockName && hasClock) {
+      // eslint-disable-next-line no-console
+      console.debug(`Renamed from ${prevClockName} to ${clockName}`);
+    }
+  }
 
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
+  componentWillUnmount() {
+    document.addEventListener('contextmenu', this.hideClock);
+    document.addEventListener('click', this.showClock);
+  }
```

Comment:
1. [CODE KNOWLEDGE] - Check if you remove your event listener, when component will unmount from page